backend/hct_mis_api/apps/account/models.py

```diff
@@ -230,28 +115,49 @@ def is_default(self) -> bool:
     def is_editable(self) -> bool:
         return not self.is_unicef and not self.is_default
 
-    def has_complete_access_in_program(self, program_id: str, business_area_id: str) -> bool:
-        return self.is_unicef or self.get_permissions().areas_for(business_area_id, program_id) == []
+    def has_full_area_access_in_program(self, program_id: Union[str, UUID]) -> bool:
+        return self.is_unicef or (
+            self.program_partner_through.filter(program_id=program_id).first()
+            and self.program_partner_through.filter(program_id=program_id).first().full_area_access
+        )
 
-    @property
-    def program_ids(self) -> List[str]:
-        return self.get_permissions().program_ids()
+    def get_program_ids_for_business_area(self, business_area_id: str) -> List[str]:
+        return [
+            str(program_id)
+            for program_id in self.programs.filter(business_area_id=business_area_id).values_list("id", flat=True)
+        ]
 
-    @property
-    def business_area_ids(self) -> List[str]:
-        return self.get_permissions().business_area_ids()
+    def has_program_access(self, program_id: Union[str, UUID]) -> bool:
+        return self.is_unicef or self.programs.filter(id=program_id).exists()
 
-    def get_program_ids_for_business_area(self, business_area_id: str) -> List[str]:
-        from hct_mis_api.apps.program.models import Program
+    def has_area_access(self, area_id: Union[str, UUID], program_id: Union[str, UUID]) -> bool:
+        return self.is_unicef or self.get_program_areas(program_id).filter(id=area_id).exists()
 
-        if self.is_unicef:
-            return [
-                str(program_id)
-                for program_id in Program.objects.filter(business_area_id=business_area_id).values_list("id", flat=True)
-            ]
-        programs_for_business_area = self.get_permissions().get_programs_for_business_area(business_area_id)
+    def get_program_areas(self, program_id: Union[str, UUID]) -> QuerySet[Area]:
```

Comment:
Why we allow program id to be str?

Reply:
Because it works for both. In the previous version (`areas_for()`) we were passing `str(program)` in many places but the typing was incorrect, stating it can be `UUID` only